cli/client.go

```diff
@@ -1197,3 +1209,45 @@ func (c *viamClient) startRobotPartShell(
 	outputLoop()
 	return nil
 }
+
+func parseLogEntryFields(fieldsRaw []*structpb.Struct) ([]zap.Field, error) {
+	parsedFields := make([]zap.Field, 0, len(fieldsRaw))
+	bytes, err := json.Marshal(fieldsRaw)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(bytes, &parsedFields); err != nil {
+		return nil, err
+	}
+	return parsedFields, nil
+}
+
+func logEntryFieldsToString(fields []zap.Field) string {
+	fieldsString := ""
+	for _, field := range fields {
+		fieldValue := ""
+		// TODO use zap encoder to encode
+		// it is a bit hard to do that though because sending the data over the wire
+		// loses type info which causes the zap encoders to panic.
+		// This code is modeled after zapcore.Field.AddTo
+		//nolint:exhaustive
+		switch field.Type {
+		case zapcore.StringType, zapcore.ErrorType:
+			fieldValue = field.String
+		case zapcore.BoolType:
+			fieldValue = fmt.Sprint(field.Integer == 1)
+		case zapcore.DurationType:
+			fieldValue = time.Duration(field.Integer).String()
+		case zapcore.Int64Type, zapcore.Int32Type, zapcore.Int16Type, zapcore.Int8Type:
+			fieldValue = fmt.Sprint(field.Integer)
+		case zapcore.Float64Type:
+			fieldValue = fmt.Sprint(math.Float64frombits(uint64(field.Integer)))
```

Comment:
Does this conversion return the bit-values or the interpreted value?
I'm not sure how the whole `float_to_log` -> `zap.Field.Int64` -> `Uint64` -> `Float64` conversion works in our setup.

Reply:
I'll talk about this in the NetCode meeting today. The conversion is unfortunately both complicated and lossy from what I understand.

Reply:
That SGTM; for posterity we're referring to https://github.com/viamrobotics/rdk/pull/3603.